sandman/source/schedule.cpp

```diff
@@ -26,42 +21,116 @@
 // Types
 //
 
-// A schedule event.
-struct ScheduleEvent
-{
-	// Read a schedule event from JSON. 
-	//
-	// p_EventObject:	The JSON object representing the event.
-	//	
-	// Returns:		True if the event was read successfully, false otherwise.
-	//
-	bool ReadFromJSON(rapidjson::Value::ConstObject const& p_EventObject);
-	
-	// Delay in seconds before this entry occurs (since the last).
-	unsigned int	m_DelaySec;
-	
-	// The control action to perform at the scheduled time.
-	ControlAction	m_ControlAction;
-};
-
 // Locals
 //
 
 // Whether the system is initialized.
 static bool s_ScheduleInitialized = false;
 
-// This will contain the schedule once it has been parsed in.
-static std::vector<ScheduleEvent> s_ScheduleEvents;
-
 // The current spot in the schedule.
 static unsigned int s_ScheduleIndex = UINT_MAX;
 
 // The time the delay for the next event began.
 static Time s_ScheduleDelayStartTime;
 
+static Schedule *s_Schedule = NULL;
+
 // Functions
 //
 
+// Schedule members
+
+Schedule::Schedule() {
+}
+
+void Schedule::AddEvent(ScheduleEvent event)
```

Comment:
This should be a constant reference rather than a copy. So ScheduleEvent const& p_Event. Note the variable name as well. I'm going to start working on some coding standards to better document this.